src/fmt.rs

```diff
@@ -0,0 +1,218 @@
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        SystemTime.format_time(&mut Writer::new(&mut timestamp))?;
+
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(IoWriter(&mut writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_entry("timestamp", &timestamp)?;
+            serializer.serialize_entry("level", &meta.level().as_serde())?;
+
+            // add all event fields to the json object
+            let mut visitor = SerdeMapVisitor::new(serializer);
+            event.record(&mut visitor);
+            serializer = visitor.take_serializer()?;
+
+            serializer.serialize_entry("target", meta.target())?;
+
+            // extract tracing information from the current span context
+            let current_span = Span::current();
+            if let Some(id) = current_span.id() {
+                let otel_ctx = current_span.context();
+                let span_ref = otel_ctx.span();
+                let span_context = span_ref.span_context();
+
+                if let Some(leaf_span) = ctx.span(&id).or_else(|| ctx.lookup_current()) {
+                    serializer.serialize_entry(
+                        "spans",
+                        &SpanScope(leaf_span, PhantomData::<N>),
+                    )?;
+                }
+
+                let trace_id = span_context.trace_id().to_string();
+                serializer.serialize_entry("trace_id", &trace_id)?;
+
+                let span_id = span_context.span_id().to_string();
+                serializer.serialize_entry("span_id", &span_id)?;
+            }
+
+            SerializeMap::end(serializer)
+        };
+
+        visit().map_err(|_| fmt::Error)?;
+        writeln!(writer)
+    }
+}
+
+struct IoWriter<'a>(&'a mut dyn fmt::Write);
+
+impl io::Write for IoWriter<'_> {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let s = str::from_utf8(buf)
+            .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
+
+        self.0.write_str(s).map_err(io::Error::other)?;
+
+        Ok(s.len())
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+}
+
+struct SpanData<'a, R, N>(SpanRef<'a, R>, PhantomData<N>)
+where
+    R: for<'lookup> LookupSpan<'lookup>,
+    N: for<'writer> FormatFields<'writer> + 'static;
+
+impl<R, N> Serialize for SpanData<'_, R, N>
+where
+    R: for<'lookup> LookupSpan<'lookup>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut serializer = serializer.serialize_map(None)?;
+        serializer.serialize_entry("name", self.0.name())?;
```

Comment:
`FormattedFields` could contain a field called "name" as well. This will output json with a duplicate key.

We could fix it, but `tracing_subscriber` have the same issue now.